crates/aptos-rosetta/src/construction.rs

```diff
@@ -247,20 +241,64 @@ async fn construction_metadata(
     }
 
     let sequence_number = if let Some(sequence_number) = request.options.sequence_number {
-        sequence_number
+        sequence_number.0
     } else {
         // Retrieve the sequence number from the rest server if one wasn't provided
         response.inner().sequence_number
     };
 
+    // Determine max gas by simulation if it isn't provided
+    let max_gas_amount = if let Some(max_gas) = request.options.max_gas_amount {
+        max_gas.0
+    } else {
+        let transaction_factory = TransactionFactory::new(server_context.chain_id)
+            .with_gas_unit_price(1)
```

Comment:
We don't need to discover this first, it'll always be 1?

Reply:
It's just estimated but let me check up on it.

The gas units should be the same regardless of price.